core/chain-configs/src/updateable_config.rs

```diff
@@ -10,7 +10,7 @@ use std::sync::{Arc, Mutex};
 /// When initializing sub-objects (e.g. `ShardsManager`), please make sure to
 /// pass this wrapper instead of passing a value from a single moment in time.
 /// See `expected_shutdown` for an example how to use it.
-#[derive(Clone)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment:
created ticket: https://pagodaplatform.atlassian.net/browse/ND-283 will also add TODO in code.

Reply:
I think currently there's issue with not including `field_name` in deserialize, because current implementation of set_metrics https://github.com/near/nearcore/blob/master/core/chain-configs/src/updateable_config.rs#L53 rely on the field_name that comes from the MutableConfigValue. If user supplies a config with an update-able field only containing the value, the set_metrics would not work. Tho all other functionalities are supposed to work if I make another `fn new(value)` that only intakes the value and fills in a default name and current timestamp to create the MutableConfigValue object.

If we want to only include value in serialize and deserialize, and we want the set_metrics to still work, we should first move the set_metrics functionality out of the implementations of MutableConfigValue type and get the name of the updated field from outer json, e.g. instead of taking `name1` from the filed_name of {"name2": {"value": 100, "field_name": "name1", "utc_time": xxx}}, we take `name2` which is the name of this mutableConfigValue field.

Reply:
but doing `expected_shutdown: 123` in config.json and `killall -SIGHUP neard` seem to already be effective at shutting down the node. I guess that means the current default serde Deserialize already handles missing field_name, but not sure what field_name the set_metrics is using 🤔

Reply:
ok I see. so when initializing client, we initialized expected_shutdown: MutableConfigValue::new(None, "expected_shutdown"),, later when user updates their config using expected_shutdown: {a number}, only the value field of expected_shutdown is modified not the field_name, so set_metrics would also be doing alright.

So only including value field in deserialize and serialize would have no issues.

Reply:
took another look at [dynamic config PR](https://github.com/near/nearcore/pull/8240/files#diff-bfdd0b5f8b3aedf5ea722418e87f6f2953822383c15c15902aed2717c886f37a ), looks like the current workflow of updating expected_shutdown does not really need a Deserializer. What happens in such workflow is: user supplies config.json, we read it in as [Config](https://github.com/near/nearcore/blob/d8a6cae167ed2f998da5504e49ef72ba5d640d5e/nearcore/src/config.rs#L348) type, where `expected_shutdown` is of type `Option<BlockHeight>`, then we use this value to update the value_field of client_config.expected_shutdown, which is of type `MutableConfigValue<Option<BlockHeight>>`. 

Deserializer for MutableConfigValue will only become useful if we want to use this type MutableConfigValue for future update-able fields in  [Config](https://github.com/near/nearcore/blob/d8a6cae167ed2f998da5504e49ef72ba5d640d5e/nearcore/src/config.rs#L348) type.

Reply:
Custom serializer: https://github.com/near/nearcore/pull/8401